File: Project Euler/p15.py
# Lattice paths
def countPath(gridSize):
    paths = 1
    for i in range(gridSize):
        paths *= (2 * gridSize) - i
        paths /= i + 1
    return int(paths)

def main():
    return countPath(20)

if __name__ == "__main__":
    print(main())

File: Project Euler/p15.cpp
// Lattice paths
#include <bits/stdc++.h>

using namespace std;

long countPath(long gridSize) {
    long paths = 1;
    for (long i = 0; i < gridSize; i++) {
        paths *= (2 * gridSize) - i;
        paths /= i + 1;
    }
    return paths;
}

int main() {
    cout << countPath(20) << endl;
    return 0;
}
